### ISSUE TITLE
Cannot determine used modules if the project is in Angular CLI

### ISSUE THREAD

**Issue by virtualSharif** [dylang/npm-check#226]
Deep mataching is not able to detect.

hence shows wrong information for "NOT USED"
<img width="1083" alt="npm-check" src="https://cloud.githubusercontent.com/assets/10231194/24234944/e6efb3da-0fc0-11e7-8b2c-392058b560b4.png">

**Comment by LinusU** [comment]
How is it used in the code?

**Comment by virtualSharif** [comment]
so here is the directory structure:
- src/app/app.module.ts
- package.json

app.module.ts file look like below : 
`
..
import { FormsModule } from '@angular/forms';
import { HttpModule } from '@angular/http';
...
....`

**Comment by LinusU** [comment]
Hmm, maybe it doesn't look in `.ts`-files 🤔 could you try temporary renaming it to `.js` and see if that solves it?

**Comment by seevee** [comment]
I tried changing the extension to .js on my app.module.ts file, but did not see any difference.

In my case, it also doesn't recognize that my @angular packages all need updates (still on 2.3.1), but does mark them all as not used.

**Comment by LinusU** [comment]
Hmm, that is very strange, could it be ignoring the files somehow? If you create a new file, `test.js`, in the root of the project and add `import { FormsModule } from '@angular/forms'`, does it complain then?

**Comment by seevee** [comment]
Strangely enough, I see the updates today. I suspect this is probably due to the angular 4.0.0 release. Unfortunately, the angular packages (minus @angular/cli) are still all show as not used. This is also the case for most of the other libraries I am actively using (d3, express, webpack loaders, etc.)

Adding the file as you suggested does remove the 'NOTUSED' status for all directly and indirectly imported libraries (importing forms resolves @angular/forms, @angular/common, @angular/core). Removing that file brings the depcheck 'NOTUSED' back.

**Comment by dhniels** [comment]
all my angular packages are shown as not used as well. I just ignore it by running npm-check -u

**Comment by ezaum** [comment]
Yeah, I have the same issue. Lots of used packages show as not used.
Project was created with angular/cli.

**Comment by isaacplmann** [comment]
I was having this same issue, but installing typescript globally fixed it for me.

```
npm i -g npm-check typescript
npm-check
```
